chapter08/ard/ex3.pl

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+use 5.010;
+
+use IO::Dir;
+
+my @dirs;
+
+foreach my $arg ( @ARGV ) {
+  if ( -d $arg ) {
+    push @dirs, $arg;
+  }
+  else {
+    say "'$arg' is not a valid directory";
+  }
+}
+
+my @dir_handles = map { my $dh = IO::Dir->new( $_ ); $dh } @dirs;
```

Comment:
Could simplify this to `map { IO::Dir->new( $_ ) } @dirs`, or even `map IO::Dir->new( $_ ), @dirs` (though the latter can be confusing).